test/k8sT/fqdn.go

```diff
@@ -0,0 +1,168 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sTest
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("K8sFQDNTest", func() {
+	var (
+		kubectl          *helpers.Kubectl
+		microscopeErr    error
+		microscopeCancel                    = func() error { return nil }
+		backgroundCancel context.CancelFunc = func() { return }
+		backgroundError  error
+
+		bindManifest = helpers.ManifestGet("bind_deployment.yaml")
+		demoManifest = helpers.ManifestGet("demo.yaml")
+
+		apps    = []string{helpers.App2, helpers.App3}
+		appPods map[string]string
+
+		worldTarget          = "http://world1.cilium.test"
+		worldTargetIP        = "192.168.9.10"
+		worldInvalidTarget   = "http://world2.cilium.test"
+		worldInvalidTargetIP = "192.168.9.11"
+	)
+
+	BeforeAll(func() {
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+		ProvisionInfraPods(kubectl)
+
+		bindManifest = helpers.ManifestGet("bind_deployment.yaml")
+
+		res := kubectl.Apply(bindManifest)
+		res.ExpectSuccess("Bind config cannot be deployed")
+
+		res = kubectl.Apply(demoManifest)
+		res.ExpectSuccess("Bind config cannot be deployed")
+
+		err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
+		Expect(err).Should(BeNil(), "Demo app is not ready after timeout")
+
+		appPods = helpers.GetAppPods(apps, helpers.DefaultNamespace, kubectl, "id")
+
+		err = kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=bind", 300)
+		Expect(err).Should(BeNil(), "Demo app is not ready after timeout")
+
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport(helpers.KubeSystemNamespace,
+			"cilium service list",
+			"cilium endpoint list")
+	})
+
+	AfterAll(func() {
+		_ = kubectl.Delete(bindManifest)
+		_ = kubectl.Delete(demoManifest)
+		ExpectAllPodsTerminated(kubectl)
+	})
+
+	JustBeforeEach(func() {
+		microscopeErr, microscopeCancel = kubectl.MicroscopeStart()
+		Expect(microscopeErr).To(BeNil(), "Microscope cannot be started")
+		backgroundCancel, backgroundError = kubectl.BackgroundReport("uptime")
+		Expect(backgroundError).To(BeNil(), "Cannot start background report process")
+	})
+
+	JustAfterEach(func() {
+		kubectl.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
+		Expect(microscopeCancel()).To(BeNil(), "cannot stop microscope")
+		backgroundCancel()
+	})
+
+	AfterEach(func() {
+		_ = kubectl.Exec(fmt.Sprintf("%s delete --all cnp", helpers.KubectlCmd))
+	})
+
+	It("Restart Cilium validate that FQDN is still working", func() {
+		connectivityTest := func() {
+			res := kubectl.ExecPodCmd(
```

Comment:
For future debugging, add `By` statements that indicate what exactly is being tested, e.g. "Testing connectivity from app2 to http://world1.cilium.test"